ViscoElasticProblem.py

```diff
@@ -137,100 +142,119 @@ def __init__(self, prob: ViscoElasticProblem, parameters: dict) -> None:
-                            (self.lambda_m_n_tableau + dt * phi_v)
-        return self.Tf_partial_next
-
+        x = SpatialCoordinate(self.mesh)
+        def fict_temp_init(x):
+            values = np.full(x.shape[1], fict_temp_value, dtype = ScalarType) 
+            return values
+        self.Tf_current.interpolate(fict_temp_init)
+        self.Tf_previous.interpolate(fict_temp_init)         
     
-    def _Tf_next(self,T_current,dt):
-        """
-        Perform weighted summation of all partial fictive temperature values.
-        C.f. Nielsen et al., eq. 26
-        """
-        # Reset for accumulation
-        Tf_expression = Expression(np.dot(self._Tf_partial_next(T_current,dt,self._phi_v(T_current)),self.m_n_tableau), self.fss.element.interpolation_points())
-        return self.Tf_next.interpolate(Tf_expression)
-
-    def _phi_v(self, T_current):
+    def _phi_v(self, T_current):     #ufl.mathfunctions.Exp
         """
         The shift function, c.f. Nielsen et al., eq. 25
         """
         return ufl.exp(
             self.H / self.Rg * (
                 1 / self.Tb -
                 self.chi / T_current -
-                (1 - self.chi) / self.Tf_current
+                (1 - self.chi) / self.Tf_previous
             )
         )
+    
+    def _Tf_partial_current(self,T_current,dt, phi_v):  #numpy.ndarray
+        """
+        Update current values for partial fictive temperature based on previous values.
+        C.f. Nielsen et al., eq. 24
+        """
+        self.Tf_partial_current = (self.lambda_m_n_tableau * self.Tf_partial_previous + T_current * dt * phi_v) / \
+                            (self.lambda_m_n_tableau + dt * phi_v)
+        return self.Tf_partial_current
 
-    def _eps_th(self,T_current,T_previous):
+    def _Tf_current(self,T_current,dt):   #NoneType - the problem is here
+        """
+        Perform weighted summation of all partial fictive temperature values.
+        C.f. Nielsen et al., eq. 26
+        """
+        # Reset for accumulation
+        Tf_expression = Expression((np.dot(self._Tf_partial_current(T_current,dt,self._phi_v(T_current)),self.m_n_tableau)), self.fss.element.interpolation_points())
+        return self.Tf_current.interpolate(Tf_expression)
+    
+    def compute_Tf_current(self,T_current,dt):  #NoneType
+        return self._Tf_current(T_current,dt)  
+
+    def _eps_th(self,T_current,T_previous):   #ufl.tensors.ComponentTensor(delta_eth)
         """
         Thermal strain tensor, c.f. Nielsen et al., eq. 9
         """
         return self.I * (
             self.alpha_solid * (T_current - T_previous)
-            - (self.alpha_liquid - self.alpha_solid) * (self.Tf_next - self.Tf_current)
+            + (self.alpha_liquid - self.alpha_solid) * (self.Tf_current - self.Tf_previous)
             )
     
-    def _strain_increment_tensor(self, T_current,T_previous):
+    def _strain_increment_tensor(self,T_current,T_previous):  #ufl.tensors.ComponentTensor(delta_eps)
```

Comment:
The comments you made here to denote the return type shouldn't be in the final commit as it doesn't conform to the python code formatting standards.

If you prefer to have it documented, you can put it in the docstring, e.g. here:

```python
"""
The total strain tensor. In absence of mechanical loads, this is trivially given.
C.f. Nielsen et al., eq. 28
Returns: ufl.tensors.ComponentTensor
"""
```
I think that it's a nice thing to have, but I'll leave that up to you